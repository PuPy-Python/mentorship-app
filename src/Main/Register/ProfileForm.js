import React from 'react';
import { formValueSelector, reduxForm, Field, FormSection } from 'redux-form';
import { compose } from 'recompose';
import { connect } from 'react-redux';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import { withRouter } from 'react-router-dom';

import RadioGroup from '../forms/RadioGroup';
import validate from './RegistrationValidation';
import TextField from '../forms/TextField';
import SelectField from '../forms/SelectField';
import { createAccount } from '../actions/registration';

const experience = [
  { label: '0-1', value: 'entry' },
  { label: '1-3', value: 'junior' },
  { label: '3-7', value: 'intermediate' },
  { label: '7+', value: 'senior' },
];

const interests = [
  { label: 'Portfolio/Code Review', value: 'portfolio' },
  { label: 'Job Search and Interviews', value: 'jobSearch' },
  { label: 'Industry Trends, Skills, Technologies', value: 'skills' },
  { label: 'Leadership, Management', value: 'leadership' },
  { label: 'Business, Entrepreneurship', value: 'business' },
  { label: 'Career Growth', value: 'careerGrowth' },
];

export const ProfileForm = ({
  handleSubmit,
  createAccount,
  classes,
  goToPrevious,
  accountType = 'mentee',
  history,
  registration,
}) => (
  <form onSubmit={handleSubmit(values => createAccount(values, history))} noValidate>
    <Typography variant="subheading" color="secondary" gutterBottom>
      CREATE YOUR {accountType.toUpperCase()} PROFILE
    </Typography>
    <Typography variant="subheading" color="secondary" gutterBottom>
      _____
    </Typography>

    <FormSection name="profile">
      <Field name="slack_handle" label="Slack Handle" component={TextField} />
      <Field name="linked_in_url" label="Linkedin URL" component={TextField} />
      <Field name="projects_url" label="Code Repository URL" component={TextField} />
      <Field name="bio" label="Bio" multiline rows="8" component={TextField} required />
    </FormSection>
    <FormSection name={accountType}>
      {(accountType === 'mentee' && (
        <Field name="goals" label="Goals" multiline rows="5" component={TextField} required />
      )) ||
        (accountType === 'mentor' && (
          <Field
            name="mentee_capacity"
            label="Mentee Capacity"
            type="number"
            component={TextField}
            required
          />
        ))}
    </FormSection>
    <FormSection name="profile">
      <Field
        name="years_industry_experience"
        label="Years of Industry Experience"
        component={RadioGroup}
        source={experience}
      />
    </FormSection>
    <FormSection name={accountType}>
      <Field
        name="areas_of_guidance"
        label="Areas of Guidance"
        multiple
        component={SelectField}
        items={interests}
        required
      />
    </FormSection>

    <Button
      variant="raised"
      color="secondary"
      size="large"
      className={classes.button}
      onClick={goToPrevious}
    >
      BACK
    </Button>
    <Button variant="raised" type="submit" color="primary" size="large" className={classes.button}>
      CONTINUE
    </Button>
    {registration.user && (
      <Typography style={{ color: 'red' }}>
        {Object.keys(registration.user).map(key => registration.user[key])}
      </Typography>
    )}
    {registration.profile && (
      <Typography style={{ color: 'red' }}>
        {Object.keys(registration.profile).map(key => registration.profile[key])}
      </Typography>
    )}
  </form>
);

ProfileForm.displayName = 'ProfileForm';

const selector = formValueSelector('registration');

const mapStateToProps = state => ({
  accountType: selector(state, 'accountType'),
  registration: state.registration,
});

const mapDispatchToProps = { createAccount };

export default compose(
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  withRouter,
  reduxForm({
    form: 'registration',
    destroyOnUnmount: false,
    forceUnregisterOnUnmount: true,
    validate,
  })
)(ProfileForm);
