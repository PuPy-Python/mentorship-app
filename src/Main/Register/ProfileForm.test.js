import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Button from '@material-ui/core/Button';

import { ProfileForm } from './ProfileForm';

const defaultProps = {
  classes: {},
  goToPrevious: jest.fn(),
  handleSubmit: jest.fn(),
};

const setup = props => shallow(<ProfileForm {...defaultProps} {...props} />);

describe('<ProfileForm />', () => {
  it('renders snapshot correctly', () => {
    const wrapper = setup();

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders the correct title when accountType = mentor', () => {
    const wrapper = setup({ accountType: 'mentor' });

    expect(wrapper.findWhere(n => n.text() === 'MENTEE')).toHaveLength(0);
    expect(wrapper.findWhere(n => n.text() === 'MENTOR')).toHaveLength(1);
  });

  it('renders the correct title when accountType = mentee', () => {
    const wrapper = setup({ accountType: 'mentee' });

    expect(wrapper.findWhere(n => n.text() === 'MENTEE')).toHaveLength(1);
    expect(wrapper.findWhere(n => n.text() === 'MENTOR')).toHaveLength(0);
  });

  it('renders mentee capacity when accountType = mentor', () => {
    const wrapper = setup({ accountType: 'mentor' });

    expect(wrapper.find('[name="mentee_capacity"]')).toHaveLength(1);
  });

  it('does not render mentee capacity when accountType = mentee', () => {
    const wrapper = setup({ accountType: 'mentee' });

    expect(wrapper.find('[name="menteeCapacity"]')).toHaveLength(0);
  });

  it('clicking the back button calls the goToPrevious prop', () => {
    const wrapper = setup();
    wrapper
      .find(Button)
      .first()
      .simulate('click');

    expect(defaultProps.goToPrevious).toHaveBeenCalledTimes(1);
  });
});
