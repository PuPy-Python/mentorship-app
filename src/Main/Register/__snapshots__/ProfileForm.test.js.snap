// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileForm /> renders snapshot correctly 1`] = `
<form
  noValidate={true}
>
  <WithStyles(Typography)
    color="secondary"
    gutterBottom={true}
    variant="subheading"
  >
    CREATE YOUR 
    MENTEE
     PROFILE
  </WithStyles(Typography)>
  <WithStyles(Typography)
    color="secondary"
    gutterBottom={true}
    variant="subheading"
  >
    _____
  </WithStyles(Typography)>
  <FormSection
    component="div"
    name="profile"
  >
    <Field
      component={[Function]}
      label="Slack Handle"
      name="slack_handle"
    />
    <Field
      component={[Function]}
      label="Linkedin URL"
      name="linked_in_url"
    />
    <Field
      component={[Function]}
      label="Code Repository URL"
      name="projects_url"
    />
    <Field
      component={[Function]}
      label="Bio"
      multiline={true}
      name="bio"
      required={true}
      rows="8"
    />
  </FormSection>
  <FormSection
    component="div"
    name="mentee"
  >
    <Field
      component={[Function]}
      label="Goals"
      multiline={true}
      name="goals"
      required={true}
      rows="5"
    />
  </FormSection>
  <FormSection
    component="div"
    name="profile"
  >
    <Field
      component={[Function]}
      label="Years of Industry Experience"
      name="years_industry_experience"
      source={
        Array [
          Object {
            "label": "0-1",
            "value": "entry",
          },
          Object {
            "label": "1-3",
            "value": "junior",
          },
          Object {
            "label": "3-7",
            "value": "intermediate",
          },
          Object {
            "label": "7+",
            "value": "senior",
          },
        ]
      }
    />
  </FormSection>
  <FormSection
    component="div"
    name="mentee"
  >
    <Field
      component={[Function]}
      items={
        Array [
          Object {
            "label": "Portfolio/Code Review",
            "value": "portfolio",
          },
          Object {
            "label": "Job Search and Interviews",
            "value": "jobSearch",
          },
          Object {
            "label": "Industry Trends, Skills, Technologies",
            "value": "skills",
          },
          Object {
            "label": "Leadership, Management",
            "value": "leadership",
          },
          Object {
            "label": "Business, Entrepreneurship",
            "value": "business",
          },
          Object {
            "label": "Career Growth",
            "value": "careerGrowth",
          },
        ]
      }
      label="Areas of Guidance"
      multiple={true}
      name="areas_of_guidance"
      required={true}
    />
  </FormSection>
  <WithStyles(Button)
    color="secondary"
    onClick={[Function]}
    size="large"
    variant="raised"
  >
    BACK
  </WithStyles(Button)>
  <WithStyles(Button)
    color="primary"
    size="large"
    type="submit"
    variant="raised"
  >
    CONTINUE
  </WithStyles(Button)>
</form>
`;
